modules/shortcodes/instagram.php

```diff
@@ -262,6 +262,28 @@ function jetpack_shortcode_instagram( $atts ) {
 		return '';
 	}
 
+	if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
+		$url_pattern = '#http(s?)://(www\.)?instagr(\.am|am\.com)/p/([^/?]+)#i';
+		preg_match( $url_pattern, $atts['url'], $matches );
+		if ( ! $matches ) {
+			return sprintf(
+				'<a href="%1$s" class="amp-wp-embed-fallback">%2$s</a>',
+				esc_url( $atts['url'] ),
+				esc_url( $atts['url'] )
+			);
+		}
+
+		$shortcode_id = end( $matches );
+		$width        = ! empty( $atts['width'] ) ? $atts['width'] : 600;
+		$height       = ! empty( $atts['height'] ) ? $atts['height'] : 600;
```

Comment:
These are the default values [from the AMP plugin](https://github.com/ampproject/amp-wp/blob/develop/includes/embeds/class-amp-instagram-embed.php#L22-L29), though feel free to comment if you'd like different defaults.